hibernate-core/src/main/java/org/hibernate/sql/results/graph/entity/internal/EntitySelectFetchInitializer.java

```diff
@@ -192,7 +189,7 @@ public void initializeInstance(RowProcessingState rowProcessingState) {
 				entityName,
 				entityIdentifier,
 				true,
-				nullable
+				referencedModelPart.isIgnoreNotFound()
```

Comment:
Is that really enough i.e. is `ignoreNotFound` set to false automatically in case of a non-nullable many-to-one without an explicit `@NotFound` annotation?

Reply:
I think so but I'll try it